### PR TITLE
[ADD] new module account_subsequence_fiscal_year

### PR DESCRIPTION
**Pull request by legalsylvain**
Fix : #1003 

This module extends the functionality of Accounting Odoo module, to
generate custom sub sequences for accounting entries, if fiscal years are not "classic".
(ie: last day = 31 / last month = 12)

without this module, in such cases (for example, for fiscal year from 01 April to 31 May)
accounting moves of the same fiscal year will not have the same numbering: Some entries will
have ``BILL/2020/`` prefix and other will have ``BILL/2021/``, that is not allowed in many countries.

That modules fixes this problem.

### CODE REVIEW

**Comment by kittiu** [comment]
Travis is red. Rebase is needed?

File: account_subsequence_fiscal_year/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+This module extends the functionality of Accounting Odoo module, to
+generate custom sub sequences for accounting entries, if fiscal years are not "classic".
+(ie: last day = 31 / last month = 12)
+
+without this module, in such cases (for example, for fiscal year from 01 April to 31 May)
```
Comment (kittiu) [inline account_subsequence_fiscal_year/readme/DESCRIPTION.rst]:
```suggestion
without this module, in such cases (for example, for fiscal year from 01 April 2020 to 31 May 2021)
```
You mean this? I just try to make sense. :)

**Comment by kittiu** [comment]
@Saran440 can you check this PR and the issue https://github.com/OCA/account-financial-tools/issues/1003
I remember you had some problem regarding the date_range before and you did something, I don't remember the PR.
Can this PR helps?

File: account_subsequence_fiscal_year/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Copyright (C) 2020 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Accounting Subsequences per Fiscal Years",
+    "summary": "Allow to create sub sequences for account moves number, based"
+    " on the fiscal years settings",
+    "version": "12.0.1.0.1",
```
Comment (Saran440) [inline account_subsequence_fiscal_year/__manifest__.py]:
```suggestion
    "version": "12.0.1.0.0",
```

**Review by Saran440 — COMMENTED** [review]
@legalsylvain This module for enterprise only? I testing with community version and can't see some fields

File: account_subsequence_fiscal_year/readme/DESCRIPTION.rst (hunk 1 above)
Comment (legalsylvain) [reply account_subsequence_fiscal_year/readme/DESCRIPTION.rst]: thanks.

**Comment by legalsylvain** [comment]
> @legalsylvain This module for enterprise only? I testing with community version and can't see some fields

Hi @Saran440 did you follow the configure section ?

**Comment by legalsylvain** [comment]
Hi @Saran440, @kittiu could you update your review ? 

CC : @quentinDupont

**Review by quentinDupont — APPROVED** [review]

**Comment by legalsylvain** [comment]
Hi @Saran440, @kittiu any news ?

Module used since several monthes in production.

**Comment by kittiu** [comment]
Functional test and code reviewed. Looks good and solve our problem too. Thank you.

Just one question, you intend to use only for some model with context["account_sequence"], right? Why not target all models?

I think it should work with SO, PO, etc.... too.

Note: please feel free to nag me, sometime, it just passed :).

**Review by kittiu — APPROVED** [review]

**Comment by legalsylvain** [comment]
> Just one question, you intend to use only for some model with context["account_sequence"], right? Why not target all models?

1) because "natively", only sequence for accounting entries are checked with ``use_date_range`` checked. (Well, I didn't checked for all sequence, but for PO, for exemple, it is not checked).

2) because AFAIK, it's a legal obligation only in the accounting area. (but maybe I'm wrong)

3) because it's my need. ;-)

But yes, we could imagine other module like ``purchase_subsequence_fiscal_year`` that makes the same thing for purchase module. Don't know.

**Comment by legalsylvain** [comment]
Hi @kittiu you could be interested by #1148 also.

**Pull request merged**